cscs-checks/prgenv/mpi.py

```diff
@@ -0,0 +1,50 @@
+import reframe as rfm
+import reframe.utility.sanity as sn
+
+
+@rfm.required_version('>=2.14')
+@rfm.parameterized_test(['single'], ['funneled'], ['serialized'], ['multiple'])
+class Mpi_InitTest(rfm.RegressionTest):
+    def __init__(self, required_thread):
+        super().__init__()
+        self.valid_systems = ['daint:gpu', 'daint:mc', 'dom:gpu', 'dom:mc']
+        self.valid_prog_environs = ['PrgEnv-cray', 'PrgEnv-gnu',
+                                    'PrgEnv-intel', 'PrgEnv-pgi']
+        self.build_system = 'SingleSource'
+        self.sourcepath = 'mpi_init_thread.cpp'
+        self.cppflags = {
+            'single':     ['-D_MPI_THREAD_SINGLE'],
+            'funneled':   ['-D_MPI_THREAD_FUNNELED'],
+            'serialized': ['-D_MPI_THREAD_SERIALIZED'],
+            'multiple':   ['-D_MPI_THREAD_MULTIPLE']
+        }
+        #self.build_system.cppflags = self.cppflags[required_thread] + ['-static']
+        self.build_system.cppflags  = self.cppflags[required_thread]
+        self.build_system.cppflags += ['-static']
+        self.time_limit = (0, 1, 0)
+        found_mpithread = sn.extractsingle(
+            r'^mpi_thread_required=\w+\s+mpi_thread_supported=(?P<resA>\w+)'
+            r'\s+mpi_thread_queried=(?P<resB>\w+)\s+(?P<resC>\d)$',
+            self.stdout, 3, int)
```

Comment:
Why do you use the `resA`, `resB` and `resC`, since you are only looking into the `resC`?